src/database/DataTypes/Requisition.js

```diff
@@ -61,6 +69,26 @@ export class Requisition extends Realm.Object {
     this.addItem(requisitionItem);
   }
 
+  createCustomerInvoice(database) {
+    if (this.isRequest || this.isFinalised) {
+      throw new Error('Cannot create invoice from Finalised or Request Requistion ');
+    }
+    if (database.objects('Transaction').
+            filtered('linkedRequisition.id = $0', this.id).length > 0) return;
+    const transaction = createRecord(database, 'CustomerInvoice',
+                                  this.otherStoreName, /* TODO: USER */);
+    this.items.forEach(requisitionItem => {
+      const transactionItem = createRecord(database, 'TransactionItem',
+                                  transaction, requisitionItem.item);
+      database.save('TransactionItem', transactionItem);
+    });
+    transaction.linkedRequisition = this;
+    this.linkedTransaction = transaction;
```

Comment:
This occurs on initial sync with an error, I assume from the post processing call.

`null is not an object (evaluating 'linkedRequisition.linkedTransaction = transaction')`